src/diffusers/loaders/unet_loader_utils.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2024 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import copy
```

Comment:
Also, remind me why are these in `unet_loader_utils.py` and not in `lora_loader_utils.py`? I assume some of these stuff also get applied to the text encoder LoRAs no?

Reply:
No, they're only applied to the unet parts. We scale the text encoder as a whole. The user can only pass a single number and it's not expanded. Therefore, these functions are not needed for it.

Now that I think of it, it's maybe not reasonable to assume the user won't want to scale the text encoder more granularly.

But I'd say we ship this PR and if there's demand, I can add support for scaling the text encoder more granularly.

WDYT?

Reply:
100 percent agreed. Thank you for walking me through this.